runtime/src/main/java/org/corfudb/runtime/view/workflows/AddNode.java

```diff
@@ -46,7 +46,12 @@ public AddNode(@NonNull String endpointToAdd, @NonNull CorfuRuntime runtime,
     protected UUID sendRequest(@NonNull ManagementClient managementClient) throws TimeoutException {
         // Bootstrap a management server first.
         Layout layout = new Layout(runtime.getLayoutView().getLayout());
-        runtime.getManagementView().bootstrapManagementServer(nodeForWorkflow, layout).join();
+        boolean bootstrapped =
+                runtime.getManagementView().bootstrapManagementServer(nodeForWorkflow, layout);
+
+        if (!bootstrapped) {
```

Comment:
minor - the variable name bootstrapped is confusing here.  If the server is bootstrapped, we log the warning.  But the check appears to be doing the opposite.  You can change the name to something else.

Reply:
Outdated